### DISCUSSION TITLE
Humana MRF solution

### DISCUSSION THREAD

**Discussion by rrootdawg** [CMSgov/price-transparency-guide#592] [General]
Has anyone tried accessing the Humana MRF solution? Does this solution align with the intent of the mandate? CSV files? Almost 500,000 files with no real way of finding anything. Index files at the end and it looks like each index is for a group and the group has 10's of thousands of file references needing to be pulled and put together.

Wondering if it's just me or other's think this kind of misses the mark on what the intent of the mandate was.

Link:
https://developers.humana.com/Resource/PCTFilesList?fileType=innetwork

**Comment by dgolden** [comment]
@rrootdawg Yes, misses the mark. Perhaps Humana is trying to be 'nice' by providing the In-Network Rate data in an easier to digest CSV format. Human is not adhering to the CMS mandated JSON schema, and ... the reason the schema exists ... generates an explosion of data CSV.GZ files (versus 'just' a few thousands if they presented proper JSON for In-Network Rate). 

For folks who wish to follow this issue, w/o having to download and inspect, I'm sharing screen shots.

Note that hundreds of thousands of CSV.GZ.  I will open the first one which shows the expected In-Network Rate fields:
![first_in_network_file_segment](https://user-images.githubusercontent.com/6385204/186221591-bd694e87-eaed-4107-a3d8-7364932e7162.jpg)

Going to the bottom of the 500K file list, I randomly selected a TOC, for "Wags Pet Clinic".  Note how it starts off referencing one of the CSV.GZ:
![wags_pet_clinic_toc_top](https://user-images.githubusercontent.com/6385204/186222154-da69aab4-ac6c-4151-aa5f-59628b8b96b0.jpg)

Now I show the bottom of this TOC. You can see how it references the 10, 427th CSV segment for "Wags":
![wags_pet_clinic_toc_bottom](https://user-images.githubusercontent.com/6385204/186222316-4f8d2f18-5468-4fcf-adab-b4f35ae9dcc8.jpg)

**Reply by rgiljohann** [reply]
Has anyone found a way to search for a specific Humana file? 
On their site (https://developers.humana.com/syntheticdata/Resource/PCTFilesList?fileType=innetwork), I only see a way to scroll through the 200 pages. You cannot even go to a specific page, just the beginning or end. This makes it impossible to find an actual file you are looking for.

**Reply by yaffsin** [reply]
The webpage make things even harder. You can use the following url to scan files.

https://developers.humana.com/syntheticdata/Resource/GetData?fileType=innetwork&sEcho=3&iColumns=3&sColumns=%2C%2C&iDisplayStart=0&iDisplayLength=1500&mDataProp_0=name&sSearch_0=&bRegex_0=false&bSearchable_0=false&mDataProp_1=modifiedDate&sSearch_1=&bRegex_1=false&bSearchable_1=false&mDataProp_2=sizeToDisplay&sSearch_2=&bRegex_2=false&bSearchable_2=false&sSearch=&bRegex=false&_

You can incrementally change "DisplayStart=0" the part of the url to get the next list. It is kind of a scrolling but you can automate it if you want. Pages return the name of folders as well, so you need to add following for them (https://developers.humana.com/syntheticdata/Resource/DownloadTOCFile?&fileName=) .

**Reply by rgiljohann** [reply]
@yaffsin Thank you, this is exactly what I was looking for.
